lib/elixir/lib/enum.ex

```diff
@@ -1438,7 +1438,8 @@ defmodule Enum do
       {[2, 4, 6], 6}
 
   """
-  @spec map_reduce(t, any, (element, any -> {any, any})) :: {any, any}
+  @spec map_reduce(t, acc, (element, acc -> {element, acc})) :: {list, acc}
+        when acc: any
```

Comment:
The thing to remember is that `acc: any` is not necessary the same as not having `acc: any`, because `when acc: any` may imply that, if one `acc` is integer, then all of them is integer.

I don't think it matters though and dialyzer applies a broader semantic anyway, so I would remove the `when` and just rely on the type.

Reply:
Awesome, agreed. I'll do that in all instances I can find in `Enum`.

Reply:
@josevalim done.